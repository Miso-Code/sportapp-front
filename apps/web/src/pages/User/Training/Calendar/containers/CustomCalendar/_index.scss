@import 'react-big-calendar/lib/sass/styles';

.training-calendar-component {
	.rbc-toolbar-label {
		@apply uppercase font-bold text-lg;
	}

	.rbc-header {
		@apply uppercase font-bold text-sm;
	}

	.rbc-toolbar button:active,
	.rbc-toolbar button.rbc-active {
		@apply bg-[#425E91] text-white;
	}

	.rbc-toolbar {
		button:hover {
			@apply bg-[#425E91] text-white;
		}

		button:focus {
			@apply bg-[#425E91] text-white;
		}
	}

	.event-custom-type {
		&-training {
			.rbc-event {
				@apply bg-[#425E91] text-white;
			}
		}

		&-event {
			.rbc-event {
				@apply bg-[#F9A825] text-white;
			}
		}

		&-session {
			.rbc-event {
				@apply bg-[#5e339a] text-white;
			}
		}
	}
}
