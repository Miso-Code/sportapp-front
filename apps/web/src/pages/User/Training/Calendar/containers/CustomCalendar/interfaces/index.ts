import { ReactNode } from 'react'
import { Event, EventWrapperProps } from 'react-big-calendar'

export interface Props {
	events: CustomEvent[]
	onSelectEvent?: (event: CustomEvent) => void
}

export enum EventTypes {
	EVENT = 'event',
	TRAINING = 'training',
	SESSION = 'session'
}

export interface CustomEvent extends Event {
	id?: string
	custom_class?: string
	description?: string
	type?: `${EventTypes}`
	warm_up?: number
	cardio?: number
	strength?: number
	cool_down?: number
	location_latitude?: number
	location_longitude?: number
	capacity?: number
}

export interface EventWrapperCustomProps
	extends EventWrapperProps<CustomEvent> {
	children?: ReactNode
}
