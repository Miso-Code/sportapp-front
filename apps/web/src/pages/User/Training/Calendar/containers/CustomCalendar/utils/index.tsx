import { EventWrapperCustomProps } from '../interfaces'

export const components = {
	eventWrapper: (props: EventWrapperCustomProps) => {
		const { event } = props
		return (
			<div
				className={`event-wrapper event-custom-type-${event.type} ${event.custom_class ?? ''}`}>
				{props.children}
			</div>
		)
	}
}
