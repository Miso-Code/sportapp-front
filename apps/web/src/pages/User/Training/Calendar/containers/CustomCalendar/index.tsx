import format from 'date-fns/format'
import getDay from 'date-fns/getDay'
import enUS from 'date-fns/locale/en-US'
import es from 'date-fns/locale/es'
import parse from 'date-fns/parse'
import startOfWeek from 'date-fns/startOfWeek'
import { Calendar, dateFnsLocalizer } from 'react-big-calendar'
import { useTranslation } from 'react-i18next'
import { Props } from './interfaces'
import { components } from './utils'

import './_index.scss'

const locales = {
	es: es,
	en: enUS
}

export default function CustomCalendar({ events, onSelectEvent }: Props) {
	const { t, i18n } = useTranslation()

	const localizer = dateFnsLocalizer({
		format,
		parse,
		startOfWeek,
		getDay,
		locales
	})

	return (
		<Calendar
			components={components}
			localizer={localizer}
			events={events}
			culture={i18n.language}
			startAccessor='start'
			endAccessor='end'
			messages={{
				allDay: t('training.allDay'),
				previous: t('training.previous'),
				next: t('training.next'),
				today: t('training.today'),
				month: t('training.month'),
				week: t('training.week'),
				day: t('training.day'),
				date: t('training.date'),
				time: t('training.hour'),
				event: t('training.event'),
				showMore: (total) => t('training.showMore', { total }),
				agenda: t('training.schedule'),
				noEventsInRange: t('training.nextEventsEmpty'),
				tomorrow: t('training.tomorrow'),
				yesterday: t('training.yesterday'),
				work_week: t('training.workWeek')
			}}
			onSelectEvent={onSelectEvent}
			className='training-calendar-component'
			style={{ maxHeight: 'calc(100vh - 200px)' }}
		/>
	)
}
