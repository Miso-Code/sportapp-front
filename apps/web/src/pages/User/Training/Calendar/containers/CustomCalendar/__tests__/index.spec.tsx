import CustomCalendar from '..'
import { render, RenderResult } from '@testing-library/react'

jest.mock('react-big-calendar', () => ({
	Calendar: () => <div>Calendar</div>,
	dateFnsLocalizer: () => jest.fn()
}))

describe('CustomCalendar', () => {
	let wrapper: RenderResult

	beforeEach(() => {
		wrapper = render(
			<CustomCalendar events={[]} onSelectEvent={jest.fn()} />
		)
	})

	it('should render', () => {
		expect(wrapper.container).toMatchSnapshot()
	})
})
