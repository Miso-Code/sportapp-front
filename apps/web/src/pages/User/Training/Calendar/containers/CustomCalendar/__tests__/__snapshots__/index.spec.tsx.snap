// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomCalendar should render 1`] = `
<div>
  <div>
    Calendar
  </div>
</div>
`;
